packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1613,14 +1613,22 @@ export class MergeTree {
                 return node.partialLengths!.getPartialLength(refSeq, clientId);
             } else {
                 const segment = node;
+                const removalInfo = this.getRemovalInfo(segment);
+
+                if(removalInfo.removedSeq !== undefined
+                    && removalInfo.removedSeq !== UnassignedSequenceNumber
+                    && removalInfo.removedSeq <= refSeq) {
+                    // this segment is a tombstone eligible for zamboni
+                    // so should never be considered, as it may not exist
+                    // on other clients
+                    return undefined;
+                }
```

Comment:
Ugh...  tough choice regarding if we should sprinkle '?? 0' through the code or do something like this:

```ts
if (len === 0 && canZamboni(seg)) {
    ...
}
```

...in the few places it matters.  Happy with whichever you think is more maintainable.

Reply:
i thought about it a bit. and i prefer to just split the meaning and force handling, rather than hoping the caller knows to differentiate the cases. there are other places where we can do slight optimization `if undefined; continue` but i decided not to couple that with this change, hence all the `??0`